torch_xla_py/xla_dist.py

```diff
@@ -18,6 +25,16 @@
 _GCE_METADATA_ENDPOINT = 'http://metadata.google.internal'
 
 
+def concat_cmd_lst(cmd_lst, delimiter=' '):
+  concat = ''
+  for i, cmd in enumerate(cmd_lst):
+    concat += cmd
```

Comment:
This does not prepend the delimiter.
I'd do something like:
```Python
def concat_cmd_list(cmd_list, delimiter=' ', quote='"'):
  concat = ''
  for cmd in cmd_list:
    if re.match('{}.*{}$'.format(quote, quote), cmd):
      token = cmd
    else:
      token = quote + cmd + quote
    if concat:
      concat += delimiter
    concat += token
  return concat
```

Reply:
Updated. Also note that I set `quote=''` inside `_build_and_run_ssh` when calling `concat_cmd_list` because if we also wrap the quotes there we get quotes around quotes (still a mess even if changing quote types). For example without setting `quote=''`:
```
"gcloud" "-q" "compute" "ssh" "--internal-ip" "--zone=europe-west4-a" "pytorchtpudistrunner@v3-128-zt8v" "--command" "'"rm" "/home/pytorchtpudistrunner/dist_training_ptxla_15.sh"'"
```

And by `build_and_run_ssh` we already have the quoted `remote_cmd` so we shouldn't need to worry about quoting the `gcloud` commands.